library/ESign/Form/Signable.php

```diff
@@ -108,33 +108,35 @@ public function isLocked()
      */
     public function getData()
     {
-        // We assume that the formdir is the same as the table suffix, 
-        // but this may not always be the case. TODO In the future, 
-        // create a list in the list_options for formdir => table mapping
-        $table = "form_".$this->_formDir;
-        if ( $this->_formDir == 'newpatient' ) {
-            $table = "form_encounter";
+      // Use default standards based on formdir value
+      // Exceptions are specified in formdir_keys list
+      $row = sqlQuery("SELECT title FROM list_options WHERE list_id = ? AND option_id = ? AND activity = 1",
+      		array('formdir_keys', $this->_formDir));
+      if (isset($row['title'])) {
+      	$excp = json_decode("{".$row['title']."}");
+      }
+      $tbl = (isset($excp->tbl) ? $excp->tbl : "form_".$this->_formDir);
+      $id = (isset($excp->id) ? $excp->id : 'id');
+      $limit = (isset($excp->limit) ? $excp->limit : 1);
+      
+      // Get form data based on key from forms table
+      $sql = sprintf("SELECT fd.* FROM %s fd
+      		INNER JOIN forms f ON fd.%s = f.form_id
+      		WHERE f.id = ?",
+      		escape_table_name($tbl), $id);
+      if ($limit <> '*') {
+      	$sql .= ' LIMIT '.$limit;
+      }
+      $rs = sqlStatement($sql, array( $this->_formId ) );
+      if (sqlNumRows($rs) == 1) { // maintain legacy hash
+      	$frs = sqlFetchArray($rs);
+      } else {
```

Comment:
Quick question on this part which seems to be a new feature to support the physical exam. Was hoping to just get a quick explanation on this. thanks, -brady